src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs

```diff
@@ -398,5 +408,15 @@ public static void ValidateCredentials(string username, string password, string
                 throw new CommandException(Strings.SourcesCommandCredentialsRequiredWithAuthTypes);
             }
         }
+
+        public static int ParseProtocolVersion(string protocolVersion)
+        {
+            if (!int.TryParse(protocolVersion, out var protocolVersionValue))
```

Comment:
With current logic customer can request invalid protocol version like `5,6` etc.
We need to validate it like below.

https://github.com/NuGet/NuGet.Client/blob/d76a117c590f8a91b844013bba7ea9b60e469aa1/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommand.cs#L195-L204

Reply:
I've added validation and defined 2, 3, 4 as valid versions  (as per <https://learn.microsoft.com/en-us/nuget/api/nuget-protocols#nuget-protocol-version-410>) 
Everything else is now treated as invalid.
I've added tests to verify that.

Reply:
Changed valid versions to `2,3` as per discussion in other thread.